codegen/src/types/mod.rs

```diff
@@ -117,6 +117,11 @@ impl<'a> TypeGenerator<'a> {
             .clone()
     }
 
+    /// Returns type registry.
+    pub fn types(&self) -> &PortableRegistry {
+        self.type_registry
+    }
```

Comment:
Nit: Instead of exposing this in this TypeGenerator, let's just pass the `PortableRegistry` into the function instead (it's accessible as something like `self.metadata.types()` in the parent function). Not a big deal though!